src/Blocks/BlockCarpet.h

```diff
@@ -50,7 +50,15 @@ class cBlockCarpetHandler:
 
 	virtual bool CanBeAt(cChunkInterface & a_ChunkInterface, const Vector3i a_RelPos, const cChunk & a_Chunk) override
 	{
-		return (a_RelPos.y > 0) && (a_Chunk.GetBlock(a_RelPos.addedY(-1)) != E_BLOCK_AIR);
+		if (!cChunkDef::IsValidHeight(a_RelPos.y - 1))
```

Comment:
Writing the same thing but with more lines is not a valuable change IMO.

Reply:
A lot of this PR seems to just be writing the same thing more verbosely.

Reply:
> A lot of this PR seems to just be writing the same thing more verbosely.

Well the goal was refactoring the CanBeAts and I do believe that more verbose code here albeit less idiomatic, more friendly to readers. Especially since most of these functions implement much logic, but if there is a preferred idiom coding style I'll go ahead and revert. Nothing was mentioned about that in the guidelines so I had the trigger loose on these changes.

Reply:
I prefer the terser code here because the check itself is quite simple.  
"The block below me cannot be air".  
You might have to read a few of these functions for that to jump out at you.

Reply:
I don't have any preference for either of these two constructions, but there is an improvement that could be made here - with @ChaosCat's code BlockBelowPos can be defined before the height check, so that DRY is maintained and there's no chance of the two heights getting out of sync in future.